### DISCUSSION TITLE
Python Environment  not found error while pushing on Git

### DISCUSSION THREAD

**Discussion by futuraflutter** [community/community#125531] [Programming Help]
### Body

I developed aa personal trained part of gemini ai model , Using python libraries  but while pushing the complete project into the git and try to host it using streamlit it shows an error that the python environment  is not found in the module. I can't install the or support the  python environment pushing into github.
![error_gitpushing](https://github.com/community/community/assets/138868186/8517637e-970c-4eb9-a521-e38a4820b761)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by MidhileshRaj** [comment]
This error shows because the used files or library is more than 100 times bigger to upload in a git hub repository.
You can not install your personal system  as a cloud platform environment in git hub , so either you push only the project without it or add a requirment.txt file which include all the shared packages inside it which is used for the project. 
Python dependencies are packages imported within the app that aren’t part of the [Python Standard  Library ](https://docs.python.org/3/library/)
To resolve the error, upload a requirements.txt file to your repo containing the following lines:{
//Like this
vega_datasets==0.9.0
pandas==1.2.5
streamlit==0.86.0
altair==4.1.0
}

Doing so will install the :point_up: dependencies before your app is run.

You can automatically generate requirements.txt file using pipreqs package.
pip install pipreqs

then just use pipreqs in your project folder terminal

**Comment by futuraflutter** [comment]
Thank You
